Dockerfile

```diff
@@ -1,31 +1,110 @@
-FROM ubuntu:xenial
+FROM ubuntu:xenial as base
 
 RUN apt-get update && apt-get -y install \
-    git \
-    net-tools \
     sudo \
-    wget \
-&& rm -rf /var/lib/apt/lists/*
-
-RUN useradd --create-home -s /bin/bash ubuntu
-RUN adduser ubuntu sudo
-RUN echo -n 'ubuntu:ubuntu' | chpasswd
+    curl \
+    locales \
+    && rm -rf /var/lib/apt/lists/*
+RUN locale-gen en_US.UTF-8
+ENV LANG=en_US.UTF-8 \
+    LANGUAGE=en_US:en \
+    LC_ALL=en_US.UTF-8
 
-# Enable passwordless sudo for users under the "sudo" group
-RUN sed -i.bkp -e \
+# Add fixuid to change permissions for bind-mounts. Set uid to same as host with -u <uid>:<guid>
+RUN addgroup --gid 1000 docker && \
+    adduser --uid 1000 --ingroup docker --home /home/docker --shell /bin/sh --disabled-password --gecos "" docker && \
+    usermod -aG sudo docker && \
+    sed -i.bkp -e \
       's/%sudo\s\+ALL=(ALL\(:ALL\)\?)\s\+ALL/%sudo ALL=NOPASSWD:ALL/g' \
       /etc/sudoers
+RUN USER=docker && \
+    GROUP=docker && \
+    curl -SsL https://github.com/boxboat/fixuid/releases/download/v0.3/fixuid-0.3-linux-amd64.tar.gz | tar -C /usr/local/bin -xzf - && \
+    chown root:root /usr/local/bin/fixuid && \
+    chmod 4755 /usr/local/bin/fixuid && \
+    mkdir -p /etc/fixuid && \
+    printf "user: $USER\ngroup: $GROUP\npaths:\n  - /service\n" > /etc/fixuid/config.yml
 
-USER ubuntu
+RUN echo "LANG=C.UTF-8" > /etc/default/locale
 
-ADD . /home/ubuntu/IncludeOS
-WORKDIR /home/ubuntu/IncludeOS
+# Docker TAG can be specified when building with --build-arg TAG=..., this is redeclared in the source-build stage
```

Comment:
Remove all the `BRANCH` and `REPO` references. That is no longer in use. (remember the comment too)